data.py

```
@@ -0,0 +1,141 @@
+    """Recursively searches all Datasets
+    from the h5 file and add at the end of the list.
+
+    Parameters
+    ----------
+    dictionary : dict
+
+        Dictionary in a dictionary with dataset due to a datagroup.
+    list_dataset : list
+
+        List with all dataset from h5py.
+    """
+    for name in dictionary:
+        if isinstance(dictionary[name], dict):
+            list_datasets(dictionary[name], list_dataset)
+
+        else:
+            list_dataset.append(dictionary[name])
+
+
+def catalog(dictionary):
+    """Creat recursively dictionary in dictionary
+    datagroup is key and value is diction witch file inside
+    if is dataset key is name dataset and value
+    is object dataset.
+
+    Parameters
+    ----------
+    dictionary : dict
+
+        Dictionary in dictionary with dataset.
+
+    Returns
+    -------
+    dictionary : dict
+
+        Dictionary in a dictionary with dataset due to a datagroup.
+    """
+    for key in dictionary.keys():
+        # name is instance datagroup
+        if isinstance(dictionary[key], h5py.Group):
+            dictionary[key] = {z: dictionary[key][z] for z in dictionary[key]}
+            # if is I create recursively dictionary with file in that group
+            catalog(dictionary[key])
+    return dictionary
+
+
+def get_data_peaks(list_dataset):
+    """Returned Dataset with peaks data from 'hitfinder/peakinfo'.
+
+    Parameters
+    ----------
+    list_dataset : list
+
+        List with dataset from h5py.
+
+    Returns
+    -------
+    dataset : The class 'h5py._hl.dataset.Dataset'
+
+        Dataset with peaks data.
+    """
+    # we are looking for the end because in LCLS H5.
+    # this datasets is near the end.
+    for dataset in list_dataset[::-1]:
+        if dataset.name == "/processing/hitfinder/peakinfo-assembled":
+            return dataset
+    print("lack Dataset /processing/hitfinder/peakinfo-assembled \
+containing peaki cheetah")
+
```

Maybe raise an error or warning if the dataset isn't found?

---

Done in: 625c3b58bf7a1764c6c3d31aa00ccfbae525be56